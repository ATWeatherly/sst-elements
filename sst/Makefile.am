# -*- Makefile -*-
#
#

AM_CPPFLAGS =  \
	-DTIXML_USE_STL \
	$(BOOST_CPPFLAGS) \
	$(ZOLTAN_CPPFLAGS)
AM_LDFLAGS = $(BOOST_LDFLAGS) $(PARMETIS_LDFLAGS) $(ZOLTAN_LDFLAGS)

SUBDIRS = libltdl .

EXTRA_DIST =

lib_LTLIBRARIES = libsst.la
libsst_la_SOURCES =
libsst_la_LIBADD = \
	$(LIBLTDL) \
	$(BOOST_SERIALIZATION_LIB) \
	$(BOOST_MPI_LIB) \
	$(BOOST_PROGRAM_OPTIONS_LIB) \
	$(ZOLTAN_LIB) \
	$(PARMETIS_LIB) \
	$(MPILIBS)

sstdir = $(includedir)/sst
dist_sst_HEADERS =
nodist_include_HEADERS =

bin_PROGRAMS = sst.x
sst_x_SOURCES =
sst_x_LDADD = libsst.la

include include/Makefile.inc
include include/sst/Makefile.inc
include tinyxml/Makefile.inc
include src/Makefile.inc
