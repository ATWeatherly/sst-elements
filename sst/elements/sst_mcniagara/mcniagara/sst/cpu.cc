// Copyright 2009-2014 Sandia Corporation. Under the terms
// of Contract DE-AC04-94AL85000 with Sandia Corporation, the U.S.
// Government retains certain rights in this software.
// 
// Copyright (c) 2009-2014, Sandia Corporation
// All rights reserved.
// 
// This file is part of the SST software package. For license
// information, see the LICENSE file in the top level directory of the
// distribution.

#include "cpu.h"

#include <sst/memEvent.h>

// bool Cpu::clock( Cycle_t current, Time_t epoch )
bool Cpu::clock( Cycle_t current )
{
    static unsigned long num_calls = 0;
    unsigned long i;
    num_calls++;
    _CPU_DBG("id=%lu cycle=%lu getCurrentSimTime=%lu num_calls=%d this=%p\n",
             Id(), current, getCurrentSimTime(), num_calls, this);
    for (i = cyclesAtLastClock; i < getCurrentSimTime(); i++)
       mcCpu->sim_cycle(i);
    cyclesAtLastClock = getCurrentSimTime();
    return false;
    
/***
    //_CPU_DBG("id=%lu getCurrentSimTime=%lu \n", Id(), current );
//     printf("In CPU::clock\n");
    MemEvent* event = NULL; 

    if (current == 100 ) primaryComponentOKToEndSim();

    if ( state == SEND ) { 
        if ( ! event ) event = new MemEvent();

        if ( who == WHO_MEM ) { 
            event->address = 0x1000; 
            who = WHO_NIC;
        } else {
            event->address = 0x10000000; 
            who = WHO_MEM;
        }

        _CPU_DBG("xxx: send a MEM event address=%#lx @ cycle %ld\n", event->address, current );
// 	mem->send( 3 * epoch, event ); 
	mem->send( (Cycle_t)3, event ); 
// 	printf("CPU::clock -> setting state to WAIT\n");
        state = WAIT;
    } else {
// 	printf("Entering state WAIT\n");
        if ( ( event = static_cast< MemEvent* >( mem->recv() ) ) ) { 
// 	    printf("Got a mem event\n");
	  _CPU_DBG("xxx: got a MEM event address=%#lx @ cycle %ld\n", event->address, current );
// 	  printf("CPU::clock -> setting state to SEND\n");
            state = SEND;
        }
    }
    return false;
***/
}

bool Cpu::memEvent( Event* event  )
{
     MemEvent *mevent;
     mevent = dynamic_cast<MemEvent*>(event);
     if (mevent)
          _CPU_DBG( "id=%lu cycle=%lu addr=%lx\n", Id(),
                getCurrentSimTime(), mevent->address );
     else
          _CPU_DBG( "id=%lu cycle=%lu\n", Id(),
                getCurrentSimTime() );
    //cpu->send( 3 * (1.0/frequency), static_cast<CompEvent*>(event) ); 
    delete event; // SHOULD I BE DOING THIS?? it seems to work...
    return false;
}

void Cpu::memoryAccess(OffCpuIF::access_mode mode,
                       long long unsigned int address,
                       long unsigned int data_size)
{
     MemEvent *event;
    _CPU_DBG( "id=%lu cycle=%lu\n", Id(),
              getCurrentSimTime() );
     // who deletes this event? Should we reuse it?
     event = new MemEvent();
     if (mode == AM_READ)
        event->type = MemEvent::MEM_LOAD;
     else
        event->type = MemEvent::MEM_STORE;
     event->address = address + Id();
     memLink->send(getCurrentSimTime(), event); 
}

void Cpu::NICAccess(OffCpuIF::access_mode mode, long unsigned int data_size)
{
    _CPU_DBG( "id=%lu cycle=%lu\n", Id(), getCurrentSimTime() );
}


extern "C" {
Cpu* cpuMcNiagaraAllocComponent( SST::ComponentId_t id, SST::Simulation* sim,
                                    SST::Params& params )
{
//     printf("cpuAllocComponent--> sim = %p\n",sim);
    return new Cpu( id, sim, params );
}
}

#if WANT_CHECKPOINT_SUPPORT
BOOST_CLASS_EXPORT(Cpu)

// BOOST_CLASS_EXPORT_TEMPLATE4( SST::EventHandler,
//                                 Cpu, bool, SST::Cycle_t, SST::Time_t )
BOOST_CLASS_EXPORT_TEMPLATE3( SST::EventHandler,
                                Cpu, bool, SST::Cycle_t)
#endif
