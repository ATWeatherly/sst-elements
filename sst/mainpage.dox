/** @mainpage

<h1> Structural Simulation Toolkit (SST) </h1>

<h2> What it does. </h2>

Understanding the performance of large-scale, capability-class , high
performance computing (HPC) systems is possibly the most significant
challenge in the development of such a system. Because it is
impractical to construct proto-type systems for evaluation at the
desired scale, architects must turn to analytical models and
simulation to guide design decisions. Historically, the architecture
community has lacked the tools needed for a reliable and integrated
evaluation of future architectures and workloads.

The Structural Simulation Toolkit aims address this problem. The SST
provides a framework for simulating large-scale HPC systems. This
simulator allows parallel simulation of large machines at multiple
levels of detail. The SST couples models for processors, memory, and
network subsystems. The SST aims, over time, to become a standard
simulation environment for designing HPC systems by helping Industry,
Academia, and the National Labs in designing and evaluating future
architectures.

<h2> Key Interfaces for Component Writers</h2>

\subpage interfaces

<h2> Install </h2>
	
<h2> Tutorials </h2>

<h2> License </h2>

@verbatim
Copyright 2009 Sandia Corporation. Under the terms
of Contract DE-AC04-94AL85000 with Sandia Corporation, the U.S.
Government retains certain rights in this software.

Copyright (c) 2009, Sandia Corporation

All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are
met:

    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.

    * Redistributions in binary form must reproduce the above
      copyright notice, this list of conditions and the following
      disclaimer in the documentation and/or other materials provided
      with the distribution.

    * Neither the name of Sandia Corporation nor the names of its
      contributors may be used to endorse or promote products derived
      from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
@endverbatim

<h2> Contact </h2>

afrodri@sandia.gov

*/

/** @page interfaces Key Interfaces

<h2>The Component</h2>

The most important class for is SST::Component, the base class from
which all simulation components inherit. At the very least, a
component writer must create a class which inherits from
SST::Component and which contains a constructor. This class should be
created in a dynamic library (.so file) and should include a C-style
function of the form <componentName>AllocComponent(), which returns a
pointer to a newly instantiated component. 

SST::Component also contains useful functions for component setup
(SST::Component::Setup()), cleanup (SST::Component::Finish()), power
reporting (SST::Component::regPowerStats()), controlling when the
simulation stops (SST::Component::registerExit() and
SST::Component::unregisterExit() ), and for handling time (such as
SST::Component::getCurrentSimTime()).

<h2>Making Event Handlers</h2>

SST components use event handling functors to handle interactions with
other components (i.e. through an SST::Event sent over a SST::Link)
and recurring events (i.e. component clock ticks). The Event Handler
Class, SST::EventHandler, is templated to create either type of event
handler by creating a functor which invokes a given member function
whenever triggered. For example: 

@verbatim
NICeventHandler = new EventHandler<proc, bool, Event *>
    (this, &proc::handle_nic_events);
...
bool proc::handle_nic_events(Event *event) {
...
}
@endverbatim

creates an event handler which calls \c proc::handle_nic_events() with
an argument of type \c Event* - the SST::Event to be processed.
Similarly,

@verbatim
clockHandler = new EventHandler< proc, bool, Cycle_t>
    ( this, &proc::preTic );
@endverbatim

creates an event handler which invokes the function \c proc::preTic()
with the current cycle time. 

<h2>Using Event Handlers</h2>

Once created, an SST::EventHandler must be registered with the
simulation core. This can be done with the SST::LinkMap::LinkAdd()
function for events coming from another component, or by
SST::Component::registerClock(), for event handlers triggered by a
clock. For example, the handlers created above could be registed in
this way:

@verbatim
LinkAdd("mem0", NICeventHandler)
registerClock( "1Ghz", clockHandler );
@endverbatim

Note that SST::Component::registerClock() can have its period or
frequency expressed in SI units in a string. The allowed units are
specified in SST::TimeLord::getTimeConverter() function.

Also note that the SST::LinkMap::LinkAdd() function does not require
an event handler if the recieving component uses the "event pull"
mechanism with SST::Link::Recv().

<h2>Links</h2>

SST::Component s use SST::Link to communicate by passing events. An
SST::Link is specified in the XML file use to configure the
simulation, and must also be added to each component which uses it by
the SST::LinkMap::LinkAdd() function. For example, 

@verbatim
<component id="processor">
 <genericProc>
   <links>
    <link id="cpu2mem">
      <params>
       <name>mem0</name>
       <lat>1 ns</lat>
      </params>
    </link>
  </links>
 </genericProc>
</component>

<component id="memory">
 <DRAMSimC>
  <links>
   <link id="cpu2mem">
    <params>
     <name>bus</name>
     <lat>1 ns</lat>
    </params>
  </link>
 </links>
 </DRAMSimC>
</component>
@endverbatim

specifies two components, a processor and a memory. These components
are connected by an SST::Link. Each \c link block contains an \c id,
a \c name, and a \c lat : 

   - The \c id is a global identifier for the SST::Link object. Since
     multiple components connect to the link ("memory" and "processor"
     in this case), each component has a \c link block and the \c id
     is used to make it clear that they are both referencing the same
     link object.

   - The \c name specifies the local name for that component to
     reference the link. It is the string passed to the
     SST::LinkMap::LinkAdd() function to make a component connect to
     that SST::Link object.

   - The \c lat specifies the latency of the link in SI units. This is
     the minimum latency for an event to be passed from one end of the
     link to another. I.e. After the event is sent from one component
     with SST::Link::Send(), it will cause the event handler on the other
     component to be invoked after at least this period of time. More
     time can be added to this delay with different varients of the
     SST::Link::Send() function.

Other commonly used SST::Link functions are:

  - SST::Link::Send(CompEvent *event) : Send an SST::Event across the link. 

  - SST::Link::Send(SimTime_t delay, CompEvent *event) : Send an
    SST::Event with an additional delay, where the delay is in units of
    the link frequency.

  - SST::Link::Send(SimTime_t delay, TimeConverter *tc, CompEvent
    *event) : Send an SST::Event with additional delay, where the
    delay is specified in units of the SST::TimeConverter object
    supplied.

  - SST::Link::Recv() : Pull a pending SST::Event from the SST::Link.
    If there is no event handler registered with the Link, this
    function should be used to gather incoming events from the link.
    (I.e. a component "pulls" events from the link, rather than having
    them "pushed" into an event handler). 

*/
