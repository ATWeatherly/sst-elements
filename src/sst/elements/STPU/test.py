import sst

# Define the simulation components
comp_stpu = sst.Component("STPU", "STPU.STPU")
comp_stpu.addParams({
    "verbose" : 1,
    "clock" : "1GHz",
    "BWPperTic" : 1,
    "STSDispatch" : 1,
    "STSParallelism" : 1
})

comp_memory = sst.Component("memory", "memHierarchy.MemController")
comp_memory.addParams({
      "debug" : 1,
      "debug_level" : 10,
      "backend.access_time" : "100 ns",
      "backing" : "malloc", 
      "clock" : "1GHz",
      "backend.mem_size" : "512MiB"
})

# Enable statistics
sst.setStatisticLoadLevel(7)
sst.setStatisticOutput("sst.statOutputConsole")
#sst.enableAllStatisticsForComponentType("memHierarchy.MemController")


# Define the simulation links
link_stpu_mem = sst.Link("link_stpu_mem")
link_stpu_mem.connect( (comp_stpu, "mem_link", "1000ps"), (comp_memory, "direct_link", "1000ps") )

