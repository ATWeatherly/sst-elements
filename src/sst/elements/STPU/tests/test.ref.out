Constructed 500 neurons with 1244 links
BWP st2001.0 to 341
BWP st2001.0 to 256
11 neurons fired @ 64
85 neurons fired @ 128
153 neurons fired @ 192
154 neurons fired @ 256
145 neurons fired @ 320
131 neurons fired @ 384
117 neurons fired @ 448
102 neurons fired @ 512
92 neurons fired @ 576
78 neurons fired @ 640
60 neurons fired @ 704
39 neurons fired @ 768
24 neurons fired @ 832
6 neurons fired @ 896
Completed 77324 neuron firings
Completed 178400 spike deliveries
 l1cache.TotalEventsReceived : Accumulator : Sum.u64 = 199030; SumSQ.u64 = 199030; Count.u64 = 199030; 
 l1cache.TotalEventsReplayed : Accumulator : Sum.u64 = 86204; SumSQ.u64 = 86204; Count.u64 = 86204; 
 l1cache.TotalNoncacheableEventsReceived : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.CacheHits : Accumulator : Sum.u64 = 157766; SumSQ.u64 = 157766; Count.u64 = 157766; 
 l1cache.GetSHit_Arrival : Accumulator : Sum.u64 = 71562; SumSQ.u64 = 71562; Count.u64 = 71562; 
 l1cache.GetXHit_Arrival : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSXHit_Arrival : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSHit_Blocked : Accumulator : Sum.u64 = 86204; SumSQ.u64 = 86204; Count.u64 = 86204; 
 l1cache.GetXHit_Blocked : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSXHit_Blocked : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.CacheMisses : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.GetSMiss_Arrival : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.GetXMiss_Arrival : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSXMiss_Arrival : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSMiss_Blocked : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetXMiss_Blocked : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSXMiss_Blocked : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetS_recv : Accumulator : Sum.u64 = 178398; SumSQ.u64 = 178398; Count.u64 = 178398; 
 l1cache.GetX_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSX_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetSResp_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.GetXResp_recv : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.PutS_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.PutM_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.PutE_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.FetchInv_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.FetchInvX_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.Inv_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.NACK_recv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.MSHR_occupancy : Accumulator : Sum.u64 = 19642047; SumSQ.u64 = 182456455; Count.u64 = 2364162; 
 l1cache.Bank_conflicts : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_E : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetS_IS : Accumulator : Sum.u64 = 4168230; SumSQ.u64 = 842097412; Count.u64 = 20632; 
 l1cache.latency_GetS_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetX_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetX_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetX_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetSX_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetSX_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.latency_GetSX_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.EventStalledForLockedCacheline : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.evict_S : Accumulator : Sum.u64 = 20600; SumSQ.u64 = 20600; Count.u64 = 20600; 
 l1cache.evict_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetS_I : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.stateEvent_GetS_S : Accumulator : Sum.u64 = 157766; SumSQ.u64 = 157766; Count.u64 = 157766; 
 l1cache.stateEvent_GetS_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetX_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetX_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetX_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetSX_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetSX_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetSX_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetSResp_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetXResp_IS : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.stateEvent_GetXResp_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_GetXResp_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Inv_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInvX_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_Fetch_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FetchInv_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLine_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_S : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_M : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_IS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_IM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_SM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineInv_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineResp_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineResp_IB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_FlushLineResp_SB : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_GetS : Accumulator : Sum.u64 = 20632; SumSQ.u64 = 20632; Count.u64 = 20632; 
 l1cache.eventSent_GetX : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_GetSX : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_PutM : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_NACK_down : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_FlushLine : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_FlushLineInv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_FetchResp : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_FetchXResp : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_AckInv : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_GetSResp : Accumulator : Sum.u64 = 178398; SumSQ.u64 = 178398; Count.u64 = 178398; 
 l1cache.eventSent_GetXResp : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_FlushLineResp : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.stateEvent_AckPut_I : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_PutS : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
 l1cache.eventSent_PutE : Accumulator : Sum.u64 = 0; SumSQ.u64 = 0; Count.u64 = 0; 
Simulation is complete, simulated time: 2.36416 ms
